src/data/mv_polynomial.lean

```diff
@@ -195,6 +211,21 @@ finsupp.induction p
   (by have : M (C 0) := h_C 0; rwa [C_0] at this)
   (assume s a p hsp ha hp, h_add _ _ (this s a) hp)
 
+theorem induction_on' {M : mv_polynomial σ α → Prop} (p : mv_polynomial σ α) :
+    (∀ (u : σ →₀ ℕ) (a : α), M (monomial u a)) →
+    (∀ (p q : mv_polynomial σ α), M p → M q → M (p + q)) → M p :=
+begin
+  intros h1 h2,
```

Comment:
If you start your proof with `intros`, it means that you could also move some parts of your target statement to the assumptions left of the `:`.